flink-test-utils-parent/flink-test-utils/src/main/java/org/apache/flink/test/util/AbstractTestBaseJUnit5.java

```diff
@@ -0,0 +1,125 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.test.util;
+
+import org.apache.flink.client.program.MiniClusterClient;
+import org.apache.flink.runtime.client.JobStatusMessage;
+import org.apache.flink.runtime.testutils.MiniClusterResourceConfiguration;
+import org.apache.flink.test.junit5.InjectClusterClient;
+import org.apache.flink.test.junit5.MiniClusterExtension;
+import org.apache.flink.util.FileUtils;
+
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.extension.RegisterExtension;
+import org.junit.jupiter.api.io.TempDir;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.File;
+import java.io.IOException;
+
+/**
+ * Base class for unit tests that run multiple tests and want to reuse the same Flink cluster. This
+ * saves a significant amount of time, since the startup and shutdown of the Flink clusters
+ * (including actor systems, etc) usually dominates the execution of the actual tests.
+ *
+ * <p>To write a unit test against this test base, simply extend it and add one or more regular test
+ * methods and retrieve the StreamExecutionEnvironment from the context:
+ *
+ * <pre>
+ *   {@literal @}Test
+ *   public void someTest() {
+ *       ExecutionEnvironment env = ExecutionEnvironment.getExecutionEnvironment();
+ *       // test code
+ *       env.execute();
+ *   }
+ *
+ *   {@literal @}Test
+ *   public void anotherTest() {
+ *       StreamExecutionEnvironment env = StreamExecutionEnvironment.getExecutionEnvironment();
+ *       // test code
+ *       env.execute();
+ *   }
+ *
+ * </pre>
+ */
+public abstract class AbstractTestBaseJUnit5 {
```

Comment:
Curious if `JUnit5` sufix is required ?

Reply:
> Curious if `JUnit5` sufix is required ?

Thanks for your review. 
This is only done to temporarily differentiate the components of Junit4.
Because this PR does not replace all the test cases of JUnit4 test suites at once.

Reply:
This PR added 3 based classes:
- AbstractTestBaseJUnit5
- JavaProgramTestBaseJUnit5
- MultipleProgramsTestBaseJUnit5

As I understand, they are base test classes, and there are too many tests will use them in the future. So I have a couple of suggestions about them:

- I prefer using the separate JIRA to do this. And this PR focues on the hadoop module.
- About the suffix, `JUnit5` doesn't make sense to me. I prefer update the old `AbstractTestBase` to `AbstractTestBaseJUnit4`, and mark it to `@Deprecated`. And then adding the `AbstractTestBase` for junit5.
  - If we add the `JUnit5` suffix for junit5, we still need to remove the suffix after all tests are upgraded to junit5.

What do you think? 

And I want to hear more thoughts from more experts of junit5. Hi @JingGe @XComp , would you mind helping take a look in your free time? thanks~

Reply:
Sounds good to me~
Could you help to split the tasks ?
I'd like do the mission follow the  steps mentioned above.
Thanks a lot.
CC @Samrat002